### PR TITLE
Remove makefile & update eslint infrastructure

### PR DESCRIPTION
**Pull request by gunjpan**
- Using makefiles messes up coverage in CI.
npm test should be vanilla mocha [--options] (ideally mocha test/unit && test/integration).
 - make is not cross platform (ie. Windows)
 - CI has issues with coverage reports when we use make *

- Update eslint infrastructure to use `eslint-config-loopback` and remove `jscs`

Connect to https://github.com/strongloop-internal/scrum-loopback/issues/872

### CODE REVIEW

**Comment by gunjpan** [comment]
@superkhau : PTAL or delegate to someone within your FA :) Thanks.

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   "main": "index.js",
   "scripts": {
     "lint": "eslint .",
-    "test": "make test",
+    "test": "mocha -G --timeout 10000",
```
Comment (superkhau) [inline package.json]: We should change this to simply `mocha`, but that is outside the scope of this PR.

File: lib/discovery.js [hunk 2]
```diff
@@ -48,6 +49,6 @@ function mixinDiscovery(Cloudant) {
     if (!options.visited.hasOwnProperty(dbname)) {
       options.visited[dbname] = schema;
     }
-    cb && cb(null, options.visited);
```
Comment (superkhau) [inline lib/discovery.js]: Is the linter complaining about using this shorthand? It looks good to me.

**Comment by superkhau** [comment]
Minor comments, LGTM. Please squash before merging.

**Comment by superkhau** [comment]
Also, please make sure CI failures are unrelated to this PR.

File: lib/cloudant.js [hunk 3]
```diff
@@ -92,7 +95,7 @@ Cloudant.prototype.toDB = function(modelName, modelObject, doc) {
     delete doc[idName];
   } else {
     if (idValue) doc._id = idValue;
-    idName !== '_id' && delete doc[idName];
```
Comment (superkhau) [inline lib/cloudant.js]: Is the linter complaining about this too?

**Comment by jannyHou** [comment]
@rmg Have we finished setting up ENV configs for cloudant on our Jenkins? From the report the DB config is not valid so all test cases fail. Thanks!

**Comment by superkhau** [comment]
@jannyHou See https://github.com/strongloop-internal/scrum-loopback/issues/1053

**Comment by superkhau** [comment]
@slnode test please

**Comment by superkhau** [comment]
@jannyHou I fixed the errors and rebased, the changes are LGTM. Please check CI and make sure the changes are unrelated before merging

**Comment by superkhau** [comment]
Make sure to squash and leave a good commit message too. ;)

**Comment by superkhau** [comment]
@gunjpan Did you backport this to loopback-2.x? This is causing CI to fail linting, PTAL http://ci.strongloop.com/job/loopback-connector-cloudant/node=0.10,os=windows/69/consoleFull#-4062861317ca194ba-b528-4e74-a19c-5de762bfc5b0. I checked locally too and it fails when you checkout `loopback-2.x` and run `npm test`

**Comment by gunjpan** [comment]
@superkhau : No I didn't. I don't remember merging it. :thinking:

**Comment by jannyHou** [comment]
@gunjpan I merged it lol.
No worries I will backport it: https://github.com/strongloop/loopback-connector-cloudant/issues/41
